test/facade/MonitoringFacade.test.ts

```diff
@@ -57,4 +63,48 @@ describe("test of defaults", () => {
       DashboardName: "testPrefix-Dynamic2",
     });
   });
+
+  test("SingleWidgetDashboardSegment is not displayed if addSegment overrides set to false", () => {
+    // configure monitoring facade with dashboard factory which populates all three default dashboards
+    const stack = new Stack();
+    const dashboardFactory = new DefaultDashboardFactory(
+      stack,
+      "TestDashboardFactory",
+      {
+        dashboardNamePrefix: "testPrefix",
+        createDashboard: true,
+        createAlarmDashboard: true,
+        createSummaryDashboard: true,
+      }
+    );
+    const facade = new MonitoringFacade(stack, "Test", {
+      dashboardFactory: dashboardFactory,
+    });
+
+    // add SingleWidgetDashboardSegment with default ctor args but use overrides to
+    // direct exclusion from default dashboards
+    facade.addSegment(
+      new SingleWidgetDashboardSegment(
+        new TextWidget({ markdown: "Simple Dashboard Segment" })
+      ),
+      {
+        addToAlarmDashboard: false,
+        addToDetailDashboard: false,
+        addToSummaryDashboard: false,
+      }
+    );
+
+    // verify that the generated dashboards do not include the SingleWidgetDashboardSegment
+    // due to override exclusion directives
+    const result = Template.fromStack(stack);
+    const dashboardCapture = new Capture();
+    result.hasResourceProperties("AWS::CloudWatch::Dashboard", {
+      DashboardBody: dashboardCapture,
+    });
+    for (var i = 0; i < 3; i++) {
```

Comment:
Is there a way to not hardcode 3? I'd be concerned about potentially missing some items.

Reply:
Yeah, I'll make this more precise.

Reply:
Converted this do a do;while loop check checks the truthiness of capture.next().